File: desafio-02/amonzada/javascript/primos.js
function ehPrimo(numero) {
    if (numero <= 1) return false;
    for (let divisor = 2; divisor <= Math.sqrt(numero); divisor++) {
        if (numero % divisor === 0) return false;
    }
    return true;
}

const numerosPrimos = [];

for (let numero = 1; numero <= 10000; numero++) {
    if (ehPrimo(numero)) {
        numerosPrimos.push(numero);
    }
}
console.log("Números primos entre 1 e 10000:");
console.log(numerosPrimos.join(", "));
console.log(`Total de números primos encontrados: ${numerosPrimos.length}`);

File: desafio-02/Richell007/java/Primos.java
public class Primos {
 
    public static void main(String[] args) {
        boolean[] ehPrimo = new boolean[10001];
 
        for (int i = 2; i <= 10000; i++) {
            ehPrimo[i] = true;
        }
 
        for (int i = 2; i * i <= 10000; i++) {
            if (ehPrimo[i]) {
                for (int j = i * i; j <= 10000; j += i) {
                    ehPrimo[j] = false;
                }
            }
        }
 
        for (int i = 2; i <= 10000; i++) {
            if (ehPrimo[i]) {
                System.out.println(i);
            }
        }
    }
}
 
